content/player/blockchain-integration/buying-nfts.md

```diff
@@ -0,0 +1,237 @@
+    address _tokenAddress,
+    uint256 _tokenId,
+    uint256 _price,
+    uint256 _duration,
+    bytes memory _fingerprint
+)
+```
+
+More information about the `fingerprint` argument can be found in [Integrating Decentraland's Estate in your Marketplace](/player/blockchain-integration/estates-marketplace-integrations)
+
+As the owner of the NFT. You can accept any Bids that have been made for it as long as it has not expired and the bidder still has enough MANA to pay for it.
+
+In order to accept a Bid, the owner of the asset has to transfer the NFT to the Bids contract using the `safeTransferFrom` function of the NFT contract with some extra data containing the accepted Bid information.
+
+For example, to accept a Bid for one of my LANDs I would have to call the LAND contract `safeTransferFrom` function to transfer my LAND to the Bids contract which will then execute the bid.
+
+```sol
+/// @param from The current owner (or operator) of the NFT.
+/// @param to The address that will receive the NFT.
+/// @param assetId The token id of the NFT to be transferred.
+/// @param userData Arbitrary data used by the recipient of the NFT to handle the transfer.
+function safeTransferFrom(
+  address from,
+  address to,
+  uint256 assetId,
+  bytes memory userData
+)
+```
+
+Let's say I want to accept a Bid with id "0xBidId" for one of the LANDs I own that has token id "100".
+
+I would have to call the `safeTransferFrom` function on the LAND contract with the following input.
+
+```js
+safeTransferFrom(
+  "0xMyAddress", // My address as I am the owner of the LAND.
+  "0xBidsContractAddress", // The address of the Bids contract in the same network.
+  "100", // The token id of my LAND.
+  "0xBidId" // The id of the Bid I want to accept.
+);
+```
+
+When the Bids contract receives the NFT, it will check in its `onERC721Received` function that the bid is not expired and the bidder has enough funds to pay for it. If everything is correct, the NFT will be transferred to the bidder and the previous owner will be paid the amount declared in the Bid.
+
+**IMPORTANT: The bidder must have approved the Bids Smart Contract to operate MANA on its behalf or the bid execution will revert when the Bids contract tries to transfer the bidders MANA to the previous owner.**
+
+## CollectionStore
+
+This smart contract can only be found on the Polygon Network. This is because Wearable and Emote Collections only exist on that chain, meaning that there is no purpose for it to be on Ethereum unless Collections were published there.
+
+When a creator publishes a collection and it is approved by the curation committee. The creator can then put the collection on sale in the marketplace, allowing other users to be able to buy the items of that collection for the price determined by the creator.
+
+When you are buying items from a collection this way, you are actually minting a new NFT that can then be commercialized with the Polygon Marketplace or Bids smart contract.
```

Comment:
```suggestion
When you buy items from a collection this way, you are minting a new NFT that can be commercialized with the Polygon Marketplace or Bids smart contract.
```

Reply:
Maybe we can mention that this is a primary or store sales and the other ones are secondary or listing sales